src/renderers/WebGLRenderer.js

```diff
@@ -1971,7 +2031,20 @@ class WebGLRenderer {
 
 				// common camera uniforms
 
+				if ( capabilities.reverseDepthBuffer ) {
+
+					toReversedProjectionMatrix( camera.projectionMatrix );
+
+				}
+
 				p_uniforms.setValue( _gl, 'projectionMatrix', camera.projectionMatrix );
+
+				if ( capabilities.reverseDepthBuffer ) {
+
+					toForwardProjectionMatrix( camera.projectionMatrix );
+
+				}
+
```

Comment:
Can I call `uniformMatrix4f` here somehow? Should be fast and clean.

Reply:
Do you mind explaining in more detail what you want to do?

Reply:
Sure. I want to avoid a copy of the matrix or touching the user camera at all. If I can update the uniform without passing an array, but a sequence of elements, I can avoid needing to cache or copy here.

```js
const m = projectionMatrix.elements;
const isPerspectiveMatrix = m[ 11 ] === - 1;

let n11 = m[ 0 ], n12 = m[ 1 ], n13 = m[ 2 ], n14 = m[ 3 ];
let n21 = m[ 4 ], n22 = m[ 5 ], n23 = m[ 6 ], n24 = m[ 7 ];
let n31 = m[ 8 ], n32 = m[ 9 ], n33 = m[ 10 ], n34 = m[ 11 ];
let n41 = m[ 12 ], n42 = m[ 13 ], n43 = m[ 14 ], n44 = m[ 15 ];

// Convert [-1, 1] to [0, 1] projection matrix
n13 = 0.5 * n13 + 0.5 * n14;
n23 = 0.5 * n23 + 0.5 * n24;
n33 = 0.5 * n33 + 0.5 * n34;
n43 = 0.5 * n43 + 0.5 * n44;

// Reverse [0, 1] projection matrix
if ( isPerspectiveMatrix ) {

	n33 = - n33 - 1;
	n43 = - n43;

} else {

	n33 = - n33;
	n43 = - n43 + 1;

}

_gl.uniformMatrix4f( location, transpose, n11, n12, n13, n14, n21, n22, n23, n24, n31, n32, n33, n34, n41, n42, n43, n44 );
```

Reply:
The uniform update routines from `WebGLUniforms` are not public so it would be good to keep the usage of `setValue()`. In this way, the approach is more consistent and I doubt there a measurable difference in performance.